core/ui/EditTemplate/tags.tid

```diff
@@ -3,39 +3,50 @@ tags: $:/tags/EditTemplate
 
 \whitespace trim
 
-\define lingo-base() $:/language/EditTemplate/
+\procedure lingo-base() $:/language/EditTemplate/
 
-\define tag-styles()
-background-color:$(backgroundColor)$;
-fill:$(foregroundColor)$;
-color:$(foregroundColor)$;
-\end
+\function tag-styles() [[background-color:$(backgroundColor)$; fill:$(foregroundColor)$; color:$(foregroundColor)$;]substitute[]]
```

Comment:
This seems a little inelegant. Perhaps we could use the new attributes for `style.backgroundColor`, `style.fill`, `style.color`?

Reply:
@Jermolene -- I have no clue what you mean here. Can you be more specific?

Reply:
@pmario instead of creating and assign a single style attribute, assign the different style properties individually using the `style.property` syntax for attributes. For example, `style.fill=<<foregroundColor>>`

Reply:
ATM the `tag-styles` function is assigned 2 times like `class=<<tag-styles>>`, so the proposed change will create duplicated code, which imo is harder to maintain than a function "string" even if it is a bit clunky.

I'll change it to the following code, which should be easier to read and to understand

```
\function tag-styles()
[[ color:$(foregroundColor)$;
   fill:$(foregroundColor)$;
   background-color:$(backgroundColor)$; ]substitute[]]
\end
```

Reply:
Hi @pmario while readability is a consideration, this is also about robustness. Using string substitution means that things can break if unexpected characters are included in the substituted strings. This is a particular risk if the substituted string is provided by the user and not subject to validation.

Reply:
hmm, The old code did use a `\define` instead of a `\function`. The concept is the same. The [tag-styles concept is 11 years old](https://github.com/Jermolene/TiddlyWiki5/blame/master/core/ui/EditTemplate/tags.tid#L8) and it seems users did not have any problems with it.

Reply:
> it seems users did not have any problems with it.

I think that if we are going to do this sort of refactoring then our primary goal should be to demonstrate best practice. Many users will use core code as the basis for their own custom solutions, so we should be wary of inadvertently encouraging them to use techniques that we know not to be robust.

I do understand that there is an alternative perspective that the textual substitution approach is preferable for pedagogic purposes because it is so conceptually simple, but I think our collective experience with SQL (which has a simpler syntax than TW/HTML/CSS) suggests that this is a dangerous area.

Reply:
I changed it as requested